### ISSUE TITLE
When setup.py file has UTF-8 BOM, Inconsistent behavior of pipenv and pip on the install command.

### ISSUE THREAD

**Issue by 0x79H** [pypa/pipenv#5661]
### Issue description

Inconsistent behavior of pipenv and pip on the setup command.
if `setup.py` has utf-8 BOM, pip can install it, but pipenv can NOT install it.

### Expected result

pipenv should install it.

### Actual result
for pip
```
>pip insatll https://github.com/hakril/PythonForWindows/archive/refs/heads/master.zip
Collecting PythonForWindows
  Using cached https://github.com/hakril/PythonForWindows/archive/refs/heads/master.zip
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
```
for pipenv
```bash
>pipenv install https://github.com/hakril/PythonForWindows/archive/refs/heads/master.zip
Installing PythonForWindows...
Resolving PythonForWindows...
Installing...
Adding PythonForWindows to Pipfile's [packages] ...
Installation Succeeded
Pipfile.lock (91d533) out of date, updating to (61f591)...
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
Locking Failed!
[    ] Locking...
Traceback (most recent call last):
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 845, in <module>
    main()
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 831, in main
    _main(
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 811, in _main
    resolve_packages(
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 772, in resolve_packages
    results = clean_results(results, resolver, project, category)
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 629, in clean_results
    entry_dict = translate_markers(entry.get_cleaned_dict(keep_outdated=False))
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 238, in get_cleaned_dict
    if self.entry.extras != self.lockfile_entry.extras:
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 601, in __getattribute__
    return super().__getattribute__(key)
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 261, in lockfile_entry
    self._lockfile_entry = self.make_requirement(self.name, self.lockfile_dict)
  File "C:\Python\Python3\lib\site-packages\pipenv\resolver.py", line 139, in make_requirement
    return Requirement.from_pipfile(name, entry)
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 2750, in from_pipfile
    r = FileRequirement.from_pipfile(name, pipfile)
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 1835, in from_pipfile
    arg_dict["setup_info"] = arg_dict["parsed_line"].setup_info
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 806, in setup_info
    self.setup_info = self.get_setup_info()
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 795, in get_setup_info
    setup_info = SetupInfo.from_ireq(self.ireq, subdir=self.subdirectory)
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1534, in from_ireq
    created = cls.create(
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1571, in create
    created.get_initial_info()
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1394, in get_initial_info
    parsed.update(self.parse_setup_py())
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1119, in parse_setup_py
    parsed = ast_parse_setup_py(self.setup_py.as_posix())
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 840, in ast_parse_setup_py
    return SetupReader.read_setup_py(Path(path), raising)
  File "C:\Python\Python3\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 164, in read_setup_py
    body = ast.parse(content).body
  File "C:\Python\Python3\lib\ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    ﻿# -*- coding: utf-8 -*-
    ^
SyntaxError: invalid non-printable character U+FEFF
```

### Steps to replicate

`pipenv install https://github.com/hakril/PythonForWindows/archive/refs/heads/master.zip`

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2023.4.21.dev0'`

Pipenv location: `'C:\\Python\\Python3\\lib\\site-packages\\pipenv'`

Python location: `'C:\\Python\\Python3\\python.exe'`

OS Name: `'nt'`

User pip version: `'23.1'`

user Python installations found:

  - `3.10.5`: `C:\Python\Python3-10-5\python.exe`
  - `3.8.10`: `C:\Python\Python38\python.exe`
  - `2.7.16`: `C:\Python\Python2-7-16\python.exe`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.10.5',
 'os_name': 'nt',
 'platform_machine': 'AMD64',
 'platform_python_implementation': 'CPython',
 'platform_release': '10',
 'platform_system': 'Windows',
 'platform_version': '10.0.22621',
 'python_full_version': '3.10.5',
 'python_version': '3.10',
 'sys_platform': 'win32'}
```

System environment variables:

  - `ALLUSERSPROFILE`
  - `AMDRMPATH`
  - `APPDATA`
  - `CLIENTNAME`
  - `COMMONPROGRAMFILES`
  - `COMMONPROGRAMFILES(X86)`
  - `COMMONPROGRAMW6432`
  - `COMPUTERNAME`
  - `COMSPEC`
  - `DRIVERDATA`
  - `EFC_5228`
  - `FPS_BROWSER_APP_PROFILE_STRING`
  - `FPS_BROWSER_USER_PROFILE_STRING`
  - `HOMEDRIVE`
  - `HOMEPATH`
  - `JAVA_HOME`
  - `LOCALAPPDATA`
  - `LOGONSERVER`
  - `NUMBER_OF_PROCESSORS`
  - `NVM_HOME`
  - `NVM_SYMLINK`
  - `ONEDRIVE`
  - `ONEDRIVECOMMERCIAL`
  - `OS`
  - `PATH`
  - `PATHEXT`
  - `POWERSHELL_DISTRIBUTION_CHANNEL`
  - `PROCESSOR_ARCHITECTURE`
  - `PROCESSOR_IDENTIFIER`
  - `PROCESSOR_LEVEL`
  - `PROCESSOR_REVISION`
  - `PROGRAMDATA`
  - `PROGRAMFILES`
  - `PROGRAMFILES(X86)`
  - `PROGRAMW6432`
  - `PROMPT`
  - `PSMODULEPATH`
  - `PUBLIC`
  - `SESSIONNAME`
  - `SYSTEMDRIVE`
  - `SYSTEMROOT`
  - `TEMP`
  - `TMP`
  - `USERDOMAIN_ROAMINGPROFILE`
  - `USERPROFILE`
  - `WINDIR`
  - `_NT_SYMBOL_PATH`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PIP_PYTHON_PATH`
  - `PYTHONDONTWRITEBYTECODE`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `C:\Program Files (x86)\VMware\VMware Workstation\bin\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;D:\Backup\msys64\usr\bin;C:\Python\Python3;C:\Python\Python3\Scripts;C:\Python\Python2;C:\Python\Python2\Scripts;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\150\Tools\Bin\;C:\Program Files\010 Editor;C:\Program Files\WireGuard\;C:\Program Files (x86)\ClockworkMod\Universal Adb Driver\;C:\Users\root\AppData\Roaming\nvm;C:\Program Files\nodejs;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\dotnet\;C:\Program Files\PowerShell\7\;C:\Program Files\BellSoft\LibericaJDK-19-Full\bin\;C:\msys64\usr\bin;C:\msys64\mingw64\bin;C:\Users\root\AppData\Local\Microsoft\WindowsApps;C:\Users\root\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\root\AppData\Local\Microsoft\WinGet\Links;C:\Users\root\.dotnet\tools;C:\Users\root\.dotnet\tools`

---------------------------

Contents of `Pipfile` ('x:\\path\\to\\Pipfile'):

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
pythonforwindows = {file = "https://github.com/hakril/PythonForWindows/archive/refs/heads/master.zip"}

[dev-packages]

[requires]
python_version = "3.10"

```

</details>

**Comment by oz123** [comment]
This is fixed once we release a new version of requirementslib and vendor it.

**Comment by matteius** [comment]
This should be fixed as of `pipenv==2023.4.29`
